pycqed/measurement/measurement_control.py

```diff
@@ -198,7 +210,10 @@ def run(self, name: str=None, exp_metadata: dict=None,
         with h5d.Data(name=self.get_measurement_name(),
                       datadir=self.datadir()) as self.data_object:
             if exp_metadata is not None:
+                self.exp_metadata = deepcopy(exp_metadata)
                 self.save_exp_metadata(exp_metadata, self.data_object)
+            else:
+                self.exp_metadata = {}
```

Comment:
Do you need this else statement? You define `self.exp_metadata` in the `__init__`